presto-main/src/main/java/com/facebook/presto/connector/system/QuerySystemTable.java

```diff
@@ -129,6 +142,34 @@ private static Block resourceGroupIdToBlock(ResourceGroupId resourceGroupId)
         return blockBuilder.build();
     }
 
+    private Block sessionPropertiesToBlock(Map<String, String> properties)
+    {
+        // Since SessionRepresentation is only created via Session systemProperties is non-null
+        // TODO: determine if SessionRepresentation should add null check for systemProperties
+        // Copied from StructuralTestUtil
+        BlockBuilder mapArrayBuilder = sessionPropertiesType.createBlockBuilder(null, properties.size());
```

Comment:
These variable names are pretty confusing to me. @haozhun you have a quick suggestion?